### ISSUE TITLE
average_addon_rating is returned as a string, not a float

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-server#7822]
### Describe the problem and steps to reproduce it:
The API returns the prop on a `User` response's `average_addon_rating` as a string but really it should be a float. I'd be great if this was set properly from the API, as right now it causes us to have to locally typecast it.

**Comment by diox** [comment]
This caused https://sentry.prod.mozaws.net/operations/olympia-prod/issues/3365259/, so we reverted it in the tag in https://github.com/mozilla/addons-server/commit/be2425e4462e2a7e2f298cd9930f15e7b76343fb

The problem is that the model field is still a `CharField`, something I failed to notice earlier. It will require more work to properly fix.

**Comment by eviljeff** [comment]
`average_addon_rating` is the average of ratings of the addons that user has developed so it'll be `None`|`''` for the majority of users.

Can we just convert it in a `SerializerMethod` field?

**Comment by diox** [comment]
We could, but I'd rather also fix the column/field on the model to be consistent

**Comment by diox** [comment]
QA: when this was first introduced it caused some old accounts with no ratings posted to not be able to log in (see mozilla/addons-frontend#4627), so please check that.

**Comment by AlexandraMoga** [comment]
@diox I've tested the login function with several accounts but I'm not sure how we could properly cover this part:


> some old accounts with no ratings posted to not be able to log in

**Comment by diox** [comment]
Well this is hard for me to tell because we don't know how those accounts ended up in that state... @vcarciu's account was affected on prod, maybe on stage and dev as well ?

**Comment by AlexandraMoga** [comment]
I have tested this with several accounts for new and older users.
I have not encountered log in issues when users had no ratings posted.
I did not have access to an old account affected by the previous login issue, so this aspect is not covered entirely.
I will mark this as verified fixed based on the results I had for the accounts I've tested with. 

Verified on AMO dev and stage with FF59, Win10x64
